src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsFrameworkCompatibility.cs

```diff
@@ -71,29 +98,46 @@ public FrameworkName GetNearest(FrameworkName targetFramework, IEnumerable<Frame
                 throw new ArgumentNullException(nameof(frameworks));
             }
 
-            var nuGetTargetFramework = NuGetFramework.ParseFrameworkName(targetFramework.ToString(), DefaultFrameworkNameProvider.Instance);
+            IEnumerable<NuGetFramework> ParseAllFrameworks(IEnumerable<FrameworkName> frameworks, string argumentName)
+            {
+                foreach (FrameworkName frameworkName in frameworks)
+                {
+                    if (frameworkName == null)
+                    {
+                        throw new ArgumentException("null item in " + argumentName);
```

Comment:
The text is a bit odd to me (null item?). I recently added the following pattern in some random places, which I'd suggest here.

I suggest adding [PropertyCannotBeNull](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Packaging/Strings.resx#L862) to VsResources.resx

then 
message: <use string above with `nameof(frameworkName)`>
paramName: `nameof(frameworks)`

Reply:
I like it, thanks. My gut reaction was that it wasn't 100% clear that an item inside the enumeration was null, but the more I thought about it, the more I think customers will get it if they see that exception.